core/corecontainers/src/dictobject_impl.cpp

```diff
@@ -320,6 +320,32 @@ ErrCode DictImpl::getSerializeId(ConstCharPtr* id) const
     return OPENDAQ_SUCCESS;
 }
 
+ErrCode DictImpl::clone(IDict** cloned)
+{
+    if (cloned == nullptr)
+    {
+        return OPENDAQ_ERR_ARGUMENT_NULL;
+    }
+
+    DictImpl* lst = new(std::nothrow) DictImpl(keyId, valueId);
+    if (lst == nullptr)
+    {
+        *cloned = nullptr;
+        return OPENDAQ_SUCCESS;
+    }
+
+    auto size = this->hashTable.size();
+    lst->hashTable.reserve(size);
+    for (const auto& item : hashTable)
+    {
+        item.first->addRef();
+        item.second->addRef();
+        lst->hashTable.insert(std::make_pair(item.first, item.second));
+    }
+
```

Comment:
If key or value is itself a mutable object such as dict or list, should it be also cloned? I think yes.